### PR TITLE
images: Update terraform version in UPI image

### PR DESCRIPTION
**Pull request by rna-afk**
Updating terraform version and the matchbox provider version to
latest to help the baremetal UPI CI PR.

### CODE REVIEW

File: images/installer/Dockerfile.upi.ci [hunk 1]
```diff
@@ -45,10 +45,10 @@ ENV CLOUDSDK_PYTHON=/usr/bin/python
 
 RUN python3 -m pip install yq
 
-ENV TERRAFORM_VERSION=0.12.24
+ENV TERRAFORM_VERSION=1.0.11
 RUN curl -O https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip && \
     unzip terraform_${TERRAFORM_VERSION}_linux_amd64.zip -d /bin/
-ENV MATCHBOX_PROVIDER_VERSION=v0.3.0
+ENV MATCHBOX_PROVIDER_VERSION=v0.5.0
 RUN curl -L -O https://github.com/poseidon/terraform-provider-matchbox/releases/download/${MATCHBOX_PROVIDER_VERSION}/terraform-provider-matchbox-${MATCHBOX_PROVIDER_VERSION}-linux-amd64.tar.gz && \
```
Comment (kirankt) [inline images/installer/Dockerfile.upi.ci:52]: We should change this to: https://github.com/poseidon/terraform-provider-matchbox/releases/download/v0.5.0/terraform-provider-matchbox_0.5.0_linux_amd64.zip and use unzip.

**Review by kirankt — COMMENTED** [review]
We need to update the matchbox download link in the Dockerfile. I've added a link to the zip file. The currently used tar.gz link doesn't exist.

File: images/installer/Dockerfile.upi.ci [hunk 2]
```diff
@@ -45,13 +45,12 @@ ENV CLOUDSDK_PYTHON=/usr/bin/python
...
+ENV TERRAFORM_VERSION=1.0.11
 RUN curl -O https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip && \
     unzip terraform_${TERRAFORM_VERSION}_linux_amd64.zip -d /bin/
-ENV MATCHBOX_PROVIDER_VERSION=v0.3.0
-RUN curl -L -O https://github.com/poseidon/terraform-provider-matchbox/releases/download/${MATCHBOX_PROVIDER_VERSION}/terraform-provider-matchbox-${MATCHBOX_PROVIDER_VERSION}-linux-amd64.tar.gz && \
-    tar xzf terraform-provider-matchbox-${MATCHBOX_PROVIDER_VERSION}-linux-amd64.tar.gz && \
-    mv terraform-provider-matchbox-${MATCHBOX_PROVIDER_VERSION}-linux-amd64/terraform-provider-matchbox /bin/terraform-provider-matchbox
+ENV MATCHBOX_PROVIDER_VERSION=v0.5.0
+RUN curl -L -O https://github.com/poseidon/terraform-provider-matchbox/releases/download/v${MATCHBOX_PROVIDER_VERSION}/terraform-provider-matchbox_${MATCHBOX_PROVIDER_VERSION}_linux_amd64.zip  && \
+    unzip terraform-provider-matchbox-${MATCHBOX_PROVIDER_VERSION}-linux-amd64.tar.gz -d /bin/
```
Comment (kirankt) [inline images/installer/Dockerfile.upi.ci:53]: Need one more fix. Please remove the tar.gz extension to the file name. Also make sure to use  underscores correctly. Thanks.

File: images/installer/Dockerfile.upi.ci [hunk 3]
```diff
@@ -45,13 +45,12 @@ ENV CLOUDSDK_PYTHON=/usr/bin/python
...
+ENV TERRAFORM_VERSION=1.0.11
 RUN curl -O https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip && \
     unzip terraform_${TERRAFORM_VERSION}_linux_amd64.zip -d /bin/
-ENV MATCHBOX_PROVIDER_VERSION=v0.3.0
-RUN curl -L -O https://github.com/poseidon/terraform-provider-matchbox/releases/download/${MATCHBOX_PROVIDER_VERSION}/terraform-provider-matchbox-${MATCHBOX_PROVIDER_VERSION}-linux-amd64.tar.gz && \
-    tar xzf terraform-provider-matchbox-${MATCHBOX_PROVIDER_VERSION}-linux-amd64.tar.gz && \
-    mv terraform-provider-matchbox-${MATCHBOX_PROVIDER_VERSION}-linux-amd64/terraform-provider-matchbox /bin/terraform-provider-matchbox
+ENV MATCHBOX_PROVIDER_VERSION=v0.5.0
+RUN curl -L -O https://github.com/poseidon/terraform-provider-matchbox/releases/download/v${MATCHBOX_PROVIDER_VERSION}/terraform-provider-matchbox-${MATCHBOX_PROVIDER_VERSION}-linux_amd64.zip  && \
+    unzip terraform-provider-matchbox-${MATCHBOX_PROVIDER_VERSION}-linux-amd64.zip -d /bin/
```
Comment (kirankt) [inline images/installer/Dockerfile.upi.ci:53]: ENV MATCHBOX_PROVIDER_VERSION=0.5.0
RUN curl -L -O https://github.com/poseidon/terraform-provider-matchbox/releases/download/v${MATCHBOX_PROVIDER_VERSION}/terraform-provider-matchbox_${MATCHBOX_PROVIDER_VERSION}_linux_amd64.zip  && \
    unzip terraform-provider-matchbox_${MATCHBOX_PROVIDER_VERSION}_linux_amd64.zip -d /bin/

**Review by kirankt — CHANGES_REQUESTED** [review]

File: images/installer/Dockerfile.upi.ci (hunk 3 above)
Comment (rna-afk) [reply images/installer/Dockerfile.upi.ci:53]: Sorry, I'm not able to fix this with my IDE due to some odd spaces being added by it and have been eyeballing this with a text editor. I'll fix this properly

Comment (kirankt) [reply images/installer/Dockerfile.upi.ci]: No worries. Pay attention to the underscores and the ${MATCHBOX_PROVIDER_VERSION} use; one place uses the prefix 'v' and another doesn't. The lines I added above in the comments work correctly.

Comment (kirankt) [reply images/installer/Dockerfile.upi.ci]: We should skip most of these tests as well. Only the image builder and UPI metal tests need to pass, IMO.

**Review by kirankt — APPROVED** [review]
/lgtm

**Pull request merged**